generators/src/main/java/com/algolia/codegen/utils/OneOfUtils.java

```diff
@@ -113,6 +113,20 @@ private static boolean isNumberType(String typeName) {
     } else if (!hasDiscriminatorA && hasDiscriminatorB) {
       return 1;
     } else {
+      // If both maps have or don't have "discriminators," compare their list lengths
+      if (hasDiscriminatorA && hasDiscriminatorB) {
+        List<?> discriminatorsA = (List<?>) mapA.get("discriminators");
+        List<?> discriminatorsB = (List<?>) mapB.get("discriminators");
+
+        // Compare the lengths of the lists
+        int lengthComparison = Integer.compare(discriminatorsB.size(), discriminatorsA.size());
+
+        if (lengthComparison != 0) {
+          return lengthComparison;
+        }
```

Comment:
_nitpick_: I think having simply this would make it work too:
```suggestion
        return discriminatorsB.size() - discriminatorsA.size();
```

Reply:
Hmm, yes, I think not being used to the language made me a bit confused, here 😆